ood-portal-generator/lib/ood_portal_generator.rb

```diff
@@ -26,5 +29,25 @@ def scl_apache?
       return false if ("#{env['ID']} #{env['ID_LIKE']}" =~ /rhel/ && env['VERSION_ID'] =~ /^8/)
       true
     end
+
+    def fqdn
+      Socket.gethostbyname(Socket.gethostname).first
+    end
+
+    def dex_user
+      Etc.getpwnam('ondemand-dex')
+      'ondemand-dex'
+    rescue ArgumentError
+      Etc.getlogin
+    end
+
+    def dex_group
+      Etc.getgrnam('ondemand-dex')
+      'ondemand-dex'
+    rescue ArgumentError
+      user = Etc.getlogin
+      gid = Etc.getpwnam(user).gid
+      Etc.getgrgid(gid).name
```

Comment:
Since we have a Gemfile and are adding other gems would ood_support be appropriate here? We basically have this code block in ood_support.

Reply:
If ood_support has this, then I'd use that instead if it's a direct replacement for this particular usage.  Looking at ood_support it doesn't appear to rescue `ArgumentError` which is what happens when the entry doesn't exist.  Don't think using that would work for what this is doing.

Reply:
`Etc.getgrgid(Process.gid).name` is a shorter alternative.

Though `Process.gid` changes if you use `newgrp` while `Etc.getpwnam(Etc.getlogin).gid` appears fixed. Which is preferred? Maybe its an edge case that does not matter.

Reply:
I think that's an edge case not to worry about. Majority of the time `ondemand-dex` will exist as majority of people will install via RPM.  I updated to use shorter alternative.